### ISSUE TITLE
Using LogsExporter with google/protobuf causes deprecation messages

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-php#1068]
**Describe your environment** 
Via `composer require google/protobuf` I have:
```
            "name": "google/protobuf",
            "version": "v4.0.0RC2",
```

**Steps to reproduce**
1. Run Collector locally with otlp/http receiver on port 4318

2. Run the following code:
```php
<?php

declare(strict_types=1);

use OpenTelemetry\API\Logs\LogRecord;
use OpenTelemetry\SDK\Sdk;
use OpenTelemetry\SDK\Common\Attribute\Attributes;
use OpenTelemetry\SDK\Common\Instrumentation\InstrumentationScopeFactory;
use OpenTelemetry\SDK\Common\Time\ClockFactory;
use OpenTelemetry\SDK\Logs\Exporter\ConsoleExporterFactory;
use OpenTelemetry\SDK\Logs\LoggerProvider;
use OpenTelemetry\SDK\Logs\Processor\SimpleLogsProcessor;
use OpenTelemetry\SDK\Logs\Processor\BatchLogsProcessor;
use OpenTelemetry\API\Common\Instrumentation\Globals;
use OpenTelemetry\Contrib\Otlp\OtlpHttpTransportFactory;
use OpenTelemetry\Contrib\Otlp\LogsExporter;

require 'vendor/autoload.php';


$transport = (new OtlpHttpTransportFactory())->create('http://localhost:4318/v1/logs', 'application/json');

$exporter = new LogsExporter($transport);

$loggerProvider = new LoggerProvider(
    new SimpleLogsProcessor($exporter),
    new InstrumentationScopeFactory(Attributes::factory())
);

Sdk::builder()
    ->setLoggerProvider($loggerProvider)
    ->setAutoShutdown(true)
    ->buildAndRegisterGlobal();


$logger = Globals::loggerProvider()->getLogger('demo', '1.0', 'http://schema.url', ['scopefoo' => 'scopebar']);

$record = (new LogRecord(['foo' => 'bar', 'baz' => 'bat', 'msg' => 'hello world']))
    ->setSeverityText('INFO')
    ->setSeverityNumber(9);

$logger->emit($record);
```

**What is the expected behavior?**
No deprecation messages.

**What is the actual behavior?**

The OTLP data is correctly exported to the Collector however I also get several deprecation messages:

```
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 234
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 124
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 141
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 213
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 242
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedField::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedField.php on line 254
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedFieldIter::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedFieldIter.php on line 84
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedFieldIter::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedFieldIter.php on line 104
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedFieldIter::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedFieldIter.php on line 94
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedFieldIter::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedFieldIter.php on line 114
[Mon Jul  3 11:10:49 2023] PHP Deprecated:  Return type of Google\Protobuf\Internal\RepeatedFieldIter::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /Users/tnajaryan/work/experiments/otel-php/vendor/google/protobuf/src/Google/Protobuf/Internal/RepeatedFieldIter.php on line 74
```

**Additional context**
Part of https://github.com/open-telemetry/community/issues/1537

**Comment by brettmc** [comment]
protobuf v4 is abandoned, AFAICT (last touched 3+ years ago). We use v3.x, and I've yesterday submitted #1067 as a fix for #1065 which should ensure you get the correct protobuf version, too.

**Comment by brettmc** [comment]
This should be resolved by #1067 (which has version requirements), and was released in https://github.com/opentelemetry-php/gen-otlp-protobuf/releases/tag/1.0.0beta6
